server/routes/testutils/appSetup.ts

```diff
@@ -51,6 +52,14 @@ export const user: PrisonUser = {
   activeCaseLoadId: 'HEI',
 }
 
+export type FlashData = {
+  errors?: ValidationError[]
+  formValues?: FlashFormValues[]
+  messages?: MoJAlert[]
+  // FIXME below needed for cancel visit route; remove when this is updated to store in req.session
+  [key: string]: string[] | ValidationError[] | FlashFormValues[] | MoJAlert[] | undefined
```

Comment:
This TypeScript fudge needed because `req.flash()` is used to pass data to cancellation confirmation page. Not ideal because page breaks if it is refreshed (because flash disappears once read). Better to use `req.session`. Will sort in a follow-on PR....